### PR TITLE
Add MaxArraySize and MaxJobCount params

### PR DESCRIPTION
**Pull request by uvNikita**

### CODE REVIEW

**Comment by uvNikita** [comment]
Looks like Travis failed for some other reason:

> fog-openstack-0.1.26 requires ruby version >= 2.2.0, which is incompatible with
the current version, ruby 2.1.9p490

**Comment by Falkor** [comment]
Nickel, thanks a lot for the contribution,

**Comment by uvNikita** [comment]
@Falkor Just noticed that this PR was not merged. Is there a reason for that, or it was an accident? :)

File: templates/slurm.conf.erb [hunk 1]
```diff
@@ -228,8 +228,9 @@ Licenses=<%= scope['slurm::licenses'] %>
 
 
 
-#MaxJobCount=5000
 #MaxStepCount=40000
+MaxArraySize=<%= scope['slurm::maxarraysize'] %>
```
Comment (Falkor) [inline templates/slurm.conf.erb:284]: I would suggest to leave the same approach as for the other parameters: if the value is the default one, leave it as comment. In this case: 

```ruby
<% if scope['slurm::maxarraysize'] == scope['slurm::params::maxarraysize'] -%>
# MaxArraySize=
<% else -%>
MaxArraySize=<%= scope['slurm::maxarraysize'] %>
<% end -%>
```

File: templates/slurm.conf.erb [hunk 2]
```diff
@@ -228,8 +228,9 @@ Licenses=<%= scope['slurm::licenses'] %>
...
+MaxJobCount=<%= scope['slurm::maxjobcount'] %>
```
Comment (Falkor) [inline templates/slurm.conf.erb:289]: As above; 

I would suggest to leave the same approach as for the other parameters: if the value is the default one, leave it as comment. In this case: 

```ruby
<% if scope['slurm::maxjobcount'] == scope['slurm::params::maxjobcount']  -%>
# MaxJobCount=
<% else -%>
MaxJobCount=<%= scope['slurm::maxjobcount'] %>
<% end -%>
```

**Review by Falkor — CHANGES_REQUESTED** [review]

**Comment by Falkor** [comment]
@uvNikita sry was closed by accident.

File: templates/slurm.conf.erb (hunk 1 above)
Comment (uvNikita) [reply templates/slurm.conf.erb:284]: Good point. I made the parameter optional now with udef as a default value. What do you think about that?

Comment (Falkor) [reply templates/slurm.conf.erb:284]: yeap, even better. Then the ERB should be adapted to simply `<% if scope['slurm::maxarraysize'] -%>` I guess...

File: templates/slurm.conf.erb (hunk 2 above)
Comment (Falkor) [reply templates/slurm.conf.erb:289]: As above, `<% if scope['slurm::maxjobcount'] -%>` ?

**Comment by uvNikita** [comment]
I fixed the check in erb and tested that everything works both when value specified and ommited. For some reason, it works correctly only if checking for nil:

```erb
<% if scope['slurm::maxarraysize'].nil? -%>`
```

**Review by Falkor — APPROVED** [review]

**Pull request merged**